### PR TITLE
[Feature] handling data skew in spark-dpp

### PR DESCRIPTION
**Pull request by sxouyang**
## What type of PR is this：
- [ ] BugFix
- [X] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes # handling data skew in spark-dpp

## Problem Summary(Required) ：

Due to unreasonable partitioning and bucketing, there is a problem of data skew. When the skew is serious, the execution of etl job will be affected. On this basis, the bucket is divided into sub-buckets to alleviate the data skew.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [X] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarrocksKeys.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarrocksKeys.java:1]: Please change the license header

Comment (sxouyang) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarrocksKeys.java:15]: The license header has been changed, thanks!

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/ParquetMergeJob.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                GroupWriteSupport.setSchema(schema, conf);
+                GroupWriteSupport writeSupport = new GroupWriteSupport();
+                ParquetWriter parquetWriter = new ParquetWriter(new Path(dstPath), writeSupport,
+                        CompressionCodecName.SNAPPY,
+                        256 * 1024 * 1024, 16 * 1024,
+                        1024 * 1024,
+                        true, false,
+                        ParquetProperties.WriterVersion.PARQUET_1_0,
+                        conf);
+                for (Path path : fileList) {
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/ParquetMergeJob.java:129]: How to ensure that the files are globally ordered?

Comment (sxouyang) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/ParquetMergeJob.java:129]: by calculating a boundary, the data can be split according to the boundary to achieve local order to global order, similar to the sortby in spark.i have two ways to re-partition, but using hash for re-partition is problematic as it does not consider the case of on sorting key having multiple values, I have already removed this method.

**Comment by wyb** [comment]
some suggestions:
1. can you give some test results before and after the optimization?
2. better to add more comments to explain the optimization step.
3. how about add a job configuration that does not enable this optimization by default, because `getBucketPartitionSize in GetBucketPartitionSizeJobHandler` needs to read the data, which may increase the task execution time. this allows user to enable this optimization if necessary.

**Pull request closed without merging**